miqa/core/tasks.py

```diff
@@ -46,34 +55,37 @@ def evaluate_data(frames_by_project):
         project = Project.objects.get(id=project_id)
         for frame_id in frame_ids:
             frame = Frame.objects.get(id=frame_id)
-            if Path(frame.raw_path).exists():
+            file_path = frame.raw_path
+            if file_path.startswith('s3://') or Path(file_path).exists():
                 eval_model_name = project.evaluation_models[[frame.scan.scan_type][0]]
                 if eval_model_name not in model_to_frames_map:
                     model_to_frames_map[eval_model_name] = []
                 model_to_frames_map[eval_model_name].append(frame)
 
-    for model_name, frame_set in model_to_frames_map.items():
-        current_model = available_evaluation_models[model_name].load()
-        results = evaluate_many(current_model, [str(frame.raw_path) for frame in frame_set])
-
-        Evaluation.objects.bulk_create(
-            [
-                Evaluation(
-                    frame=frame,
-                    evaluation_model=model_name,
-                    results=results[str(frame.raw_path)],
-                )
-                for frame in frame_set
-            ]
-        )
-
-
-def _download_from_s3(path: str) -> bytes:
-    bucket, key = path.strip()[5:].split('/', maxsplit=1)
-    client = boto3.client('s3')
-    buf = BytesIO()
-    client.download_fileobj(bucket, key, buf)
-    return buf.getvalue()
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        for model_name, frame_set in model_to_frames_map.items():
+            current_model = available_evaluation_models[model_name].load()
+            file_paths = []
+            for frame in frame_set:
+                file_path = str(frame.raw_path)
+                if file_path.startswith('s3://'):
+                    tmp = tempfile.NamedTemporaryFile(prefix=tmpdirname)
+                    tmp.write(_download_from_s3(file_path))
+                    file_path = tmp.name
+                else:
```

Comment:
There should be no else. Otherwise we need something like `file_paths.append(tmp.name)` on line 74.

Reply:
Nice catch. Changed in [79415e6](https://github.com/OpenImaging/miqa/pull/363/commits/79415e6e8bd6f7d73c44cf05fc17152e5bbeb118)